### PR TITLE
[bots] Autolabeler: label to label 

### PR DESCRIPTION
**Pull request by clee2000**
Pt1 of https://github.com/pytorch/pytorch/issues/117051 but doesn't handle the topic: not user facing thing

Reads a config to add labels based on other labels added.

Config currently only supports if any -> add, if all -> add
Ex if you say if any of (A, B, C), then add (D), then D gets added if an issue or PR gets B

Only adds based on the currently added label in an attempt to prevent concurrency problems

Planning on testing on malfet/deleteme

TIL: if you add a label while making a PR or issue, two webhooks get issued: one for the PR/issue creation, one for the labeling event

Example config:
```
- any:
  - "module: custom operators"
  - "module: aotdispatch"
  then:
  - "module: pt2-dispatcher"
- any:
  - "module: dynamo"
  - "module: pt2-dispatcher"
  - "module: inductor"
  - "module: custom operators"
  then:
  - "oncall: pt2"
 ```

### CODE REVIEW

File: torchci/lib/bot/autoLabelBot.ts [hunk 1]
```diff
@@ -249,8 +314,55 @@ function myBot(app: Probot): void {
...
+    const filtered = await filterCIFlowLabels(true, newLabels);
+    if (filtered.length) {
+      await addLabels(context, filtered);
+    }
+  });
+
+  app.on("pull_request.labeled", async (context) => {
+    // Careful! Read the above comments about label addition
+
+    const label = context.payload.label!.name;
```
Comment (huydhn) [inline torchci/lib/bot/autoLabelBot.ts:337]: Nit: Could we rename them to `addedLabel` and `existingLabels` like how you name them in getLabelsFromLabelToLabelConfig?  My mental map is lying to me by telling label is a member of labels lol

Comment (huydhn) [reply torchci/lib/bot/autoLabelBot.ts:337]: Maybe doing the same for issues.labeled event too for consistency

File: torchci/lib/bot/autoLabelBot.ts [hunk 2]
```diff
@@ -249,8 +314,55 @@ function myBot(app: Probot): void {
...
+
+    const filtered = await filterCIFlowLabels(
+      false,
+      newLabels,
+      context,
+      context.payload.repository.owner.login,
+      context.payload.repository.name
+    );
+    if (filtered.length) {
+      await addLabels(context, filtered);
```
Comment (huydhn) [inline torchci/lib/bot/autoLabelBot.ts:365]: Just curious, do we want to have a check here to see if filtered is the same as the list of existing labels?  I'm thinking about a hypothetical case when label A adds label B adds label A, etc. to infinity :)

File: torchci/lib/bot/autoLabelBot.ts [hunk 3]
```diff
@@ -165,6 +167,32 @@ export async function getLabelsFromLabelerConfig(
   return labels;
 }
 
+export async function getLabelsFromLabelToLabelConfig(
```
Comment (huydhn) [inline torchci/lib/bot/autoLabelBot.ts:170]: Do you have an example of label2label config at hand so I could take a look?  I check malfet/deleteme repo, but I don't think it's there yet

**Review by huydhn — APPROVED** [review]
Overall LGTM!

File: torchci/lib/bot/autoLabelBot.ts (hunk 1 above)
Comment (clee2000) [reply torchci/lib/bot/autoLabelBot.ts:337]: did this for pull request labeled, I want to refactor this file so I'll do the changes for other funcs in a different PR

File: torchci/lib/bot/autoLabelBot.ts (hunk 2 above)
Comment (clee2000) [reply torchci/lib/bot/autoLabelBot.ts:365]: Just tested, if label A gets added but already existed on the issue, nothing changes/no new webhook gets sent

File: torchci/lib/bot/autoLabelBot.ts (hunk 3 above)
Comment (clee2000) [reply torchci/lib/bot/autoLabelBot.ts:170]: The tests should have an example

**Pull request merged**